packages/flutter_goldens_client/lib/client.dart

```diff
@@ -46,17 +46,36 @@ class GoldensClient {
   /// subprocesses.
   final ProcessManager process;
 
-  RandomAccessFile _lock;
-
   /// The local [Directory] where the Flutter repository is hosted.
   ///
   /// Uses the [fs] file system.
   Directory get flutterRoot => fs.directory(platform.environment[_kFlutterRootKey]);
 
-  /// The local [Directory] where the goldens repository is hosted.
+  /// The local [Directory] where the goldens files are located.
   ///
   /// Uses the [fs] file system.
-  Directory get repositoryRoot => flutterRoot.childDirectory(fs.path.join('bin', 'cache', 'pkg', 'goldens'));
+  Directory get comparisonRoot => flutterRoot.childDirectory(fs.path.join('bin', 'cache', 'pkg', 'goldens'));
+}
+
+/// A class that represents a clone of the https://github.com/flutter/goldens
+/// repository, nested within the `bin/cache` directory of the caller's Flutter
+/// repository.
+class GoldensRepositoryClient  extends GoldensClient {
+
+  GoldensRepositoryClient() : super();
```

Comment:
Why the split into two constructors? Why can this one not just take fs, platform, process as optional arguments?

Reply:
See comment below.